stl/inc/functional

```diff
@@ -1479,6 +1479,83 @@ namespace placeholders {
     _INLINE_VAR constexpr _Ph<20> _20{};
 } // namespace placeholders
 
+#if _HAS_CXX20
+// FUNCTION TEMPLATE _Call_front_binder
+template <size_t... _Ix, class _Cv_FD, class _Cv_tuple_TiD, class... _Unbound>
+constexpr auto _Call_front_binder(index_sequence<_Ix...>, _Cv_FD&& _Obj, _Cv_tuple_TiD&& _Tpl,
+    _Unbound&&... _Unbargs) noexcept(noexcept(_STD invoke(_STD forward<_Cv_FD>(_Obj),
+    _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))..., _STD forward<_Unbound>(_Unbargs)...)))
+    -> decltype(_STD invoke(_STD forward<_Cv_FD>(_Obj), _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))...,
+        _STD forward<_Unbound>(_Unbargs)...)) {
+    return _STD invoke(_STD forward<_Cv_FD>(_Obj), _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))...,
+        _STD forward<_Unbound>(_Unbargs)...);
+}
+
+// CLASS TEMPLATE _Front_binder
+template <class _Fx, class... _Types>
+class _Front_binder { // wrap bound callable object and arguments
+private:
+    static_assert(is_constructible_v<decay_t<_Fx>, _Fx>,
+        "std::bind_front() requires the decayed callable to be constructible from an undecayed callable");
+    static_assert(is_move_constructible_v<decay_t<_Fx>>,
+        "std::bind_front() requires the decayed callable to be move constructible");
+    static_assert(conjunction_v<is_constructible<decay_t<_Types>, _Types>...>,
+        "std::bind_front() requires the decayed bound arguments to be constructible from undecayed bound arguments");
+    static_assert(conjunction_v<is_move_constructible<decay_t<_Types>>...>,
+        "std::bind_front() requires the decayed bound arguments to be move constructible");
+
+    using _Seq    = index_sequence_for<_Types...>;
+    using _First  = decay_t<_Fx>;
+    using _Second = tuple<decay_t<_Types>...>;
+
+    _Compressed_pair<_First, _Second> _Mypair;
+
+public:
+    constexpr explicit _Front_binder(_Fx&& _Func, _Types&&... _Args)
+        : _Mypair(_One_then_variadic_args_t(), _STD forward<_Fx>(_Func), _STD forward<_Types>(_Args)...) {}
+
+    template <class... _Unbound>
+        _NODISCARD constexpr auto operator()(_Unbound&&... _Unbargs)
+        & noexcept(noexcept(_Call_front_binder(_Seq{}, _Mypair._Get_first(), _Mypair._Myval2,
+            _STD forward<_Unbound>(_Unbargs)...))) -> decltype(_Call_front_binder(_Seq{}, _Mypair._Get_first(),
+            _Mypair._Myval2, _STD forward<_Unbound>(_Unbargs)...)) {
+        return _Call_front_binder(_Seq{}, _Mypair._Get_first(), _Mypair._Myval2, _STD forward<_Unbound>(_Unbargs)...);
+    }

```

Comment:
I believe that you need to update `_Compressed_pair` to be `constexpr` if you want this to be usable in constant expressions. See: https://github.com/microsoft/STL/blob/957fe99f41748b80c3108447e5dd5ce2be962f4e/stl/inc/xmemory#L1318-L1367

You should be able to do this unconditionally, as this is internal machinery and I don't believe that this will disrupt C++14/17 codepaths.